### PR TITLE
use @maplibre/maplibre-gl-style-spec / remove maplibre-gl-js submodules

### LINKED ISSUE #389: Can't update style-spec because missing rollup folder
Since the rollup folder was remove in https://github.com/maplibre/maplibre-gl-js/pull/1986, we created a lot of broken references in the docs and lost the ability to update the style spec.

### PR DESCRIPTION
**Pull request by birkskyum**
Closes #389

This uses the new style-spec npm package instead of the maplibre-gl-js submodule.

This PR updates the style spec to the latest version 18.0.1, because it's the npm package version installed.

### CODE REVIEW

File: docs/components/__tests__/expression-metadata.test.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { types } from '../expression-metadata';
-import ref from '../../../maplibre-gl-js/rollup/build/tsc/src/style-spec/reference/latest';
+import ref from '@maplibre/maplibre-gl-style-spec/tsc/src/reference/latest';
```
Comment (HarelM) [inline docs/components/__tests__/expression-metadata.test.js:2]: Is there another way to import this? Maybe from the main file somehow? It is exported as v8, not sure it has all the relevant fields though...

**Comment by HarelM** [comment]
There's also a need to update the readme with the removal of the submodule I believe...

**Comment by birkskyum** [comment]
This does not complete the task, because the expressions are nagging, but it does narrow down the issue quite a lot and it compiles.

File: docs/components/__tests__/expression-metadata.test.js (hunk 1 above)
Comment (birkskyum) [reply docs/components/__tests__/expression-metadata.test.js:2]: At this point, getting it to run without a submodule is quite a feat in itself

**Comment by birkskyum** [comment]
There are some optimizations that can be made when the docs get split and get closer to the source code.

**Review by HarelM — APPROVED** [review]

**Pull request merged**